crates/walrus-service/src/node/storage.rs

```diff
@@ -461,10 +461,11 @@ impl Storage {
     pub async fn list_shard_status(&self) -> HashMap<ShardIndex, Option<ShardStatus>> {
         let shards = self.shards.read().await;
 
-        shards
-            .iter()
-            .map(|(shard, storage)| (*shard, storage.status().ok()))
-            .collect()
+        let mut result = HashMap::new();
+        for (shard, storage) in shards.iter() {
+            result.insert(*shard, storage.status().await.ok());
+        }
+        result
```

Comment:
Remark: I felt this should also work with an iterator-like approach, but because we want a hash map, it turned out to be a bit tedious 🤷:
```rust
        futures::future::join_all(
            shards
                .iter()
                .map(|(shard, storage)| async move { (*shard, storage.status().await.ok()) }),
        )
        .await
        .into_iter()
        .collect()
```